bigquery/storage/managedwriter/client.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package managedwriter
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+	"strings"
+
+	storage "cloud.google.com/go/bigquery/storage/apiv1beta2"
+	"google.golang.org/api/option"
+	storagepb "google.golang.org/genproto/googleapis/cloud/bigquery/storage/v1beta2"
+)
+
+// Client is a managed BigQuery Storage write client scoped to a single project.
+type Client struct {
+	rawClient *storage.BigQueryWriteClient
+	projectID string
+}
+
+// NewClient instantiates a new client.
+func NewClient(ctx context.Context, projectID string, opts ...option.ClientOption) (c *Client, err error) {
+	numConns := runtime.GOMAXPROCS(0)
+	if numConns > 4 {
+		numConns = 4
+	}
+	o := []option.ClientOption{
+		option.WithGRPCConnectionPool(numConns),
+	}
+	o = append(o, opts...)
+
+	rawClient, err := storage.NewBigQueryWriteClient(ctx, o...)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{
+		rawClient: rawClient,
+		projectID: projectID,
+	}, nil
+}
+
+// NewManagedStream establishes a new stream for writing.
+//
+// destinationTable should be of the format: projects/{project}/datasets/{dataset}/tables/{table}
```

Comment:
Is destinationTable required? Should it be a param vs an optional option?

Reply:
Stale comment from refactor, corrected.